src/wallet/fees.cpp

```diff
@@ -62,13 +62,7 @@ CAmount GetMinimumFee(unsigned int nTxBytes, const CCoinControl& coin_control, c
         }
     }
 
-    // prevent user from paying a fee below minRelayTxFee or minTxFee
-    CAmount required_fee = GetRequiredFee(nTxBytes);
-    if (required_fee > fee_needed) {
-        fee_needed = required_fee;
-        if (feeCalc) feeCalc->reason = FeeReason::REQUIRED;
-    }
-    // But always obey the maximum
+    // And always obey the maximum
```

Comment:
You'd still need to set it to minTxFee in case the current fee is smaller than that (Also set the FeeReason appropriately)

Also, I'd suggest to move GetMinFee down, so that it is applied when paytxfee is set.